admin/timescale/create_indexes.sql

```diff
@@ -3,6 +3,7 @@ BEGIN;
 CREATE INDEX listened_at_user_name_ndx_listen ON listen (listened_at DESC, user_name);
 CREATE INDEX created_ndx_listen ON listen (created);
 CREATE UNIQUE INDEX listened_at_track_name_user_name_ndx_listen ON listen (listened_at DESC, track_name, user_name);
+CREATE INDEX recording_msid_ndx_listen on listen ((data->'track_metadata'->'additional_info'->>'recording_msid'));
```

Comment:
Assuming this is for https://github.com/metabrainz/listenbrainz-server/blob/3f24a5309480c52c9c872c63a56d18419ce9c56c/listenbrainz/mbid_mapping_writer/job_queue.py#L153 

Adding this index is probably fine but if postgres decides to do a hash join, the index probably won't improve performance. So may want to check execution plans before and after creating the index.

Reply:
This index is less for the current PR and more for the other things I often do in working with the mapping. For instance, in order to select MSIDs to recheck, I need to query on recording_msid frequently and its slow without an index. Much faster with one -- and in some cases the queries become near instant.